integration/spark/src/main/scala/org/apache/spark/sql/execution/command/index/DropIndexCommand.scala

```diff
@@ -184,10 +184,12 @@ private[sql] case class DropIndexCommand(
         parentCarbonTable = getRefreshedParentTable(sparkSession, dbName)
         val indexMetadata = parentCarbonTable.getIndexMetadata
         if (null != indexMetadata && null != indexMetadata.getIndexesMap) {
-          val hasCgFgIndexes =
-            !(indexMetadata.getIndexesMap.size() == 1 &&
-              indexMetadata.getIndexesMap.containsKey(IndexType.SI.getIndexProviderName))
-          if (hasCgFgIndexes) {
+          // check if any CG or FG index exists. If not exists,
+          // then set indexExists as false to return empty index list for next query.
+          val hasCgFgIndexes = indexMetadata.getIndexesMap.size() != 0 &&
```

Comment:
`indexMetadata.getIndexesMap.size() != 0` would always be true at this point. `indexMetadata` will be null if empty.

Reply:
indexmetadata will also have SI Indexes. indexExists property is only for CG or FG indexes

Reply:
Got it.

Reply:
ok, removed the check because indexes map can have max 3 elements and it won't throw any exception for empty map.